worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -371,17 +371,30 @@ function isEmptyMap(mapValue) {
     return jsonString === '{}';
 }
 
+function searchFieldsForValue(fieldValues, value) {
+    for (var fieldValue of fieldValues) {
+        if (fieldValue.isStringValue() && fieldValue.getStringValue() === value) {
+            return true;
+        }
+    }
+    return false;
+}
+
 function fieldHasStringValue(document, fieldName, value) {
 
     var fieldValues = document.getField(fieldName).getValues();
-    if (fieldValues)
-    {
-        for (var fieldValue of fieldValues) {
-            if (fieldValue.isStringValue() && fieldValue.getStringValue() === value) {
-                return true;
-            }
+
+    return searchFieldsForValue(fieldValues, value);
+}
+
+function fieldHasAnyStringValue(document, fieldName, values) {
```

Comment:
This has the same issue as yesterday.  The outer loop should be looping over the actual field values.  This is looping over the set of values being searched for in the outer loop.

Reply:
Apologies, my fault for working from another file to sample those outer loop changes. They should now be integrated into this PR @dermot-hardy

Reply:
https://github.com/CAFDataProcessing/worker-workflow/pull/98/commits/670b93cc203ef33084c938138d174559b61ec035